Model/Product/Variation/Collection.php

```diff
@@ -81,6 +81,11 @@ public function build(Product $product, NostoProduct $nostoProduct, Store $store
         $collection = new VariationCollection();
         $groups = $this->customerGroupManager->getLoggedInGroups();
         foreach ($groups as $group) {
+            // For some (broken?) Magento setups the default group / default
+            // variation is also part of the customer groups
+            if ($group->getCode() === $nostoProduct->getVariationId()) {
```

Comment:
Cast `getVariationId` to string just to be sure, annotation on `getVariationId` says it returns `mixed|null` and `getCode` returns `string`. Maybe if someone creates a group named 1 we will have a false negative.